### PR TITLE
Add guide on setting up Postgres streaming replication

### PR DESCRIPTION
**Pull request by treydock**
## Description

Add section to Postgres guide on setting up a Standby host using streaming replication.

## Motivation and Context

Improves Postgres guide to include more real-world setups.

### CODE REVIEW

**Comment by treydock** [comment]
CC @ghoneycutt

File: content/sensu-go/5.17/guides/scale-event-storage.md [hunk 1]
```diff
@@ -178,8 +179,86 @@ You may see old events in the web UI or sensuctl output until the etcd datastore
...
+30318/30318 kB (100%), 1/1 tablespace
+# Remove configurations that are only for the primary
+sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/data/postgresql.conf
+# Enable read-only queries, modify/remove if master wal_level is archive
+echo 'hot_standby = on' | sudo tee -a /var/lib/pgsql/data/postgresql.conf
+# start the postgresql service
+sudo systemcl start postgresql
+{{< /highlight >}}
+
+With this configuration complete, your Sensu events will be replicated to the standby host.
```
Comment (ghoneycutt) [inline content/sensu-go/5.17/guides/scale-event-storage.md:325]: Should add the psql commands that let you verify replication is happening.

File: content/sensu-go/5.17/guides/scale-event-storage.md [hunk 2]
```diff
@@ -178,8 +179,86 @@ You may see old events in the web UI or sensuctl output until the etcd datastore
 
 _**NOTE**: If your Sensu Go license expires, event storage will automatically revert to etcd._
 
+## Configure Postgres Streaming Replication
```
Comment (ghoneycutt) [inline content/sensu-go/5.17/guides/scale-event-storage.md]: Note this is for EL7. The commands might change a bit depending on platform.

**Comment by ghoneycutt** [comment]
@cwjohnston @agoddard

File: content/sensu-go/5.17/guides/scale-event-storage.md (hunk 2 above)
Comment (treydock) [reply content/sensu-go/5.17/guides/scale-event-storage.md]: Added a note that the paths and service names are OS dependent and may vary based on operating systems

File: content/sensu-go/5.17/guides/scale-event-storage.md (hunk 1 above)
Comment (treydock) [reply content/sensu-go/5.17/guides/scale-event-storage.md:325]: Added the commands.

File: content/sensu-go/5.19/guides/scale-event-storage.md [hunk 3]
```diff
@@ -174,7 +175,173 @@ To verify that the change was effective, look for messages similar to these in t
 Similar to enabling Postgres, switching back to the etcd datastore does not migrate current event data from one store to another.
 You may see old events in the web UI or sensuctl output until the etcd datastore catches up with the current state of your monitored infrastructure.
 
+{{% notice note %}}
+**NOTE**: If your Sensu Go license expires, event storage will automatically revert to etcd.
+{{% /notice %}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.19/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.18/guides/scale-event-storage.md [hunk 4]
```diff
@@ -174,7 +175,173 @@ To verify that the change was effective, look for messages similar to these in t
 Similar to enabling Postgres, switching back to the etcd datastore does not migrate current event data from one store to another.
 You may see old events in the web UI or sensuctl output until the etcd datastore catches up with the current state of your monitored infrastructure.
 
+{{% notice note %}}
+**NOTE**: If your Sensu Go license expires, event storage will automatically revert to etcd.
+{{% /notice %}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 5]
```diff
@@ -174,8 +175,166 @@ To verify that the change was effective, look for messages similar to these in t
 Similar to enabling Postgres, switching back to the etcd datastore does not migrate current event data from one store to another.
 You may see old events in the web UI or sensuctl output until the etcd datastore catches up with the current state of your monitored infrastructure.
 
+_**NOTE**: If your Sensu Go license expires, event storage will automatically revert to etcd._
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 6]
```diff
@@ -174,8 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
 You may see old events in the web UI or sensuctl output until the etcd datastore catches up with the current state of your monitored infrastructure.
 
+## Configure Postgres streaming replication
+
+Postgres supports an active standby by using [streaming replication][6].
+All Sensu events written to the primary Postgres server will be replicated to the standby server.
+
+_**NOTE**: Paths and service names may vary based on your operating system._
+
+This section describes how to configure PostgreSQL streaming replication in three steps.
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/guides/scale-event-storage.md]:
```suggestion
This section describes how to configure PostgreSQL streaming replication in four steps.
```

File: content/sensu-go/5.17/guides/scale-event-storage.md [hunk 7]
```diff
@@ -174,9 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+## Configure Postgres streaming replication
+
+Postgres supports an active standby by using [streaming replication][6].
+All Sensu events written to the primary Postgres server will be replicated to the standby server.
+
+{{% notice note %}}
+**NOTE**: Paths and service names may vary based on your operating system.
+{{% /notice %}}
+
+This section describes how to configure PostgreSQL streaming replication in three steps.
```
Comment (hillaryfraley) [inline content/sensu-go/5.17/guides/scale-event-storage.md]:
```suggestion
This section describes how to configure PostgreSQL streaming replication in four steps.
```

File: content/sensu-go/5.18/guides/scale-event-storage.md [hunk 8]
```diff
@@ -174,7 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+## Configure Postgres streaming replication
+
+Postgres supports an active standby by using [streaming replication][6].
+All Sensu events written to the primary Postgres server will be replicated to the standby server.
+
+{{% notice note %}}
+**NOTE**: Paths and service names may vary based on your operating system.
+{{% /notice %}}
+
+This section describes how to configure PostgreSQL streaming replication in three steps.
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/scale-event-storage.md]:
```suggestion
This section describes how to configure PostgreSQL streaming replication in four steps.
```

File: content/sensu-go/5.19/guides/scale-event-storage.md [hunk 9]
```diff
@@ -174,7 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+## Configure Postgres streaming replication
+
+Postgres supports an active standby by using [streaming replication][6].
+All Sensu events written to the primary Postgres server will be replicated to the standby server.
+
+{{% notice note %}}
+**NOTE**: Paths and service names may vary based on your operating system.
+{{% /notice %}}
+
+This section describes how to configure PostgreSQL streaming replication in three steps.
```
Comment (hillaryfraley) [inline content/sensu-go/5.19/guides/scale-event-storage.md]:
```suggestion
This section describes how to configure PostgreSQL streaming replication in four steps.
```

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 10]
```diff
@@ -174,8 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
+The next step is to set the streaming replication configuration parameters on the primary Postgres host.
+Begin by making a copy of the `postgresql.conf`:
+
+{{< highlight shell >}}
+sudo cp -a /var/lib/pgsql/data/postgresql.conf /var/lib/pgsql/data/postgresql.conf.bak
+{{< /highlight >}}
+
+Next, append the necessary configuration options.
+
+_**NOTE**: To enable read-only queries on a standby server, you must set `wal_level` to `hot_standby`.
```
Comment (cwjohnston) [inline content/sensu-go/5.16/guides/scale-event-storage.md]: Sensu does not provide means to make use of read-only replicas, so I think we should remove this note.

**Review by cwjohnston — APPROVED** [review]
Approved with one suggestion

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 11]
```diff
@@ -174,8 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
+If you will never connect to the server in standby mode, you can set `wal_level` to `archive` instead._
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.17/guides/scale-event-storage.md [hunk 12]
```diff
@@ -174,9 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+sudo cp -a /var/lib/pgsql/data/postgresql.conf /var/lib/pgsql/data/postgresql.conf.bak
+{{< /highlight >}}
+
+Next, append the necessary configuration options.
+
+{{% notice note %}}
+**NOTE**: To enable read-only queries on a standby server, you must set `wal_level` to `hot_standby`.
+If you will never connect to the server in standby mode, you can set `wal_level` to `archive` instead.
+{{% /notice %}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.17/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.18/guides/scale-event-storage.md [hunk 13]
```diff
@@ -174,7 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+sudo cp -a /var/lib/pgsql/data/postgresql.conf /var/lib/pgsql/data/postgresql.conf.bak
+{{< /highlight >}}
+
+Next, append the necessary configuration options.
+
+{{% notice note %}}
+**NOTE**: To enable read-only queries on a standby server, you must set `wal_level` to `hot_standby`.
+If you will never connect to the server in standby mode, you can set `wal_level` to `archive` instead.
+{{% /notice %}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.19/guides/scale-event-storage.md [hunk 14]
```diff
@@ -174,7 +175,169 @@ To verify that the change was effective, look for messages similar to these in t
...
+sudo cp -a /var/lib/pgsql/data/postgresql.conf /var/lib/pgsql/data/postgresql.conf.bak
+{{< /highlight >}}
+
+Next, append the necessary configuration options.
+
+{{% notice note %}}
+**NOTE**: To enable read-only queries on a standby server, you must set `wal_level` to `hot_standby`.
+If you will never connect to the server in standby mode, you can set `wal_level` to `archive` instead.
+{{% /notice %}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.19/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 15]
```diff
@@ -174,8 +175,162 @@ To verify that the change was effective, look for messages similar to these in t
...
+The next step is to set the streaming replication configuration parameters on the primary Postgres host.
+Begin by making a copy of the `postgresql.conf`:
+
+{{< highlight shell >}}
+sudo cp -a /var/lib/pgsql/data/postgresql.conf /var/lib/pgsql/data/postgresql.conf.bak
+{{< /highlight >}}
+
+Next, append the necessary configuration options.
+
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.16/guides/scale-event-storage.md [hunk 16]
```diff
@@ -174,8 +175,162 @@ To verify that the change was effective, look for messages similar to these in t
...
+{{< highlight shell >}}
+sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
+Enable read-only queries (or modify/remove, if master `wal_level` is archived):
+
+{{< highlight shell >}}
+echo 'hot_standby = on' | sudo tee -a /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.16/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.17/guides/scale-event-storage.md [hunk 17]
```diff
@@ -174,9 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
+{{< highlight shell >}}
+sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
+Enable read-only queries (or modify/remove, if master `wal_level` is archived):
+
+{{< highlight shell >}}
+echo 'hot_standby = on' | sudo tee -a /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.17/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.18/guides/scale-event-storage.md [hunk 18]
```diff
@@ -174,7 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
+{{< highlight shell >}}
+sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
+Enable read-only queries (or modify/remove, if master `wal_level` is archived):
+
+{{< highlight shell >}}
+echo 'hot_standby = on' | sudo tee -a /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/scale-event-storage.md]:
```suggestion
```

File: content/sensu-go/5.19/guides/scale-event-storage.md [hunk 19]
```diff
@@ -174,7 +175,164 @@ To verify that the change was effective, look for messages similar to these in t
...
+{{< highlight shell >}}
+sudo sed -r -i.bak '/^(wal_level|max_wal_senders|wal_keep_segments).*/d' /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
+Enable read-only queries (or modify/remove, if master `wal_level` is archived):
+
+{{< highlight shell >}}
+echo 'hot_standby = on' | sudo tee -a /var/lib/pgsql/data/postgresql.conf
+{{< /highlight >}}
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.19/guides/scale-event-storage.md]:
```suggestion
```

**Pull request merged**